cylc/flow/network/__init__.py

```diff
@@ -43,7 +43,10 @@
 
 def encode_(message):
     """Convert the structure holding a message field from JSON to a string."""
-    return json.dumps(message)
+    try:
+        return json.dumps(message)
+    except Exception as exc:
+        return json.dumps({'data': str(exc)})
```

Comment:
Was trying to avoid `json.dumps` freezing up the scheduler, I can be more specific with the error too..

Reply:
> Was trying to avoid json.dumps freezing up the scheduler

That's good but we need to send back an error message in the format which clients are expecting i.e. `{'errors': [{'message': str(exc)}]}`.